src/dhclientv4_wrapper.c

```diff
@@ -0,0 +1,279 @@
+            return;
+        }
+    }
+}
+
+static void run_dhclient(const int argc, char *argv[])
+{
+    struct sigaction sigact = {
+        .sa_handler = signal_handler,
+        .sa_flags   = 0,
+    };
+
+    /* Make sure the udhcpc has ian effective root permission to run before going farther */
+    force_identity();
+
+    /* Set up a pipe for notifying the parent's polling loop of a SIGCHLD. */
+    if (pipe(exit_pipe_fd) < 0) {
+        err(EXIT_FAILURE, "Unable to set-up the inter-process pipe!");
+    }
+
+    /* Capture SIGCHLD and other relevant signals */
+    if(sigemptyset(&sigact.sa_mask) != 0) {
+        err(EXIT_FAILURE, "Unable to set the signal mask! errno = %d; err = '%s'!", errno, strerror(errno));
+    }
+
+    if(sigaction(SIGCHLD, &sigact, NULL) != 0) {
+        err(EXIT_FAILURE, "Unable to set-up handler for a SIGCHLD signal: errno = %d; err = '%s'!", errno, strerror(errno));
+    }
+
+    if(sigaction(SIGINT,  &sigact, NULL) != 0) {
+        err(EXIT_FAILURE, "Unable to set-up handler for a SIGCINT signal: errno = %d; err = '%s'!", errno, strerror(errno));
+    }
+
+    /* Fork */
+    child_pid = fork(); /* TODO: consider using vfork so we would not have to close various file descriptors but we would have to call execve instead of execv */
+
+    if (child_pid < 0)
+        err(EXIT_FAILURE, "fork");
+
+    if (child_pid == 0) {
+        child(argc, argv);
+    } else {
+        parent();
+    }
+
+}
+
+static const char * getenv_nonull(const char * restrict key)
+{
+    const char * restrict result = getenv(key);
+    return result != NULL ? result : "";
+}
+
+static char *get_ip_addr(char * restrict dest, char * restrict ip_netmask, char * restrict ip_address, char * restrict ip_netmasklen)
+{
+    if (ip_netmask != NULL) {
+      strncpy(dest, ip_netmask, INET_ADDRSTRLEN);
+    } else if((ip_address != NULL) && (ip_netmasklen != NULL)) {
+        snprintf(dest, INET_ADDRSTRLEN, "%s/%s", ip_address, ip_netmasklen);
+    }
+
+    return dest; /* in DA60 format */
+}
+
+/* The Dhclient's environment variables input to the script:
+ * reason
+ * interface
+ * new_ip_address, ip_prefixlen - if no new_prefix avaial
+ * new_prefix
+ * new_domain_search
+ * new_domain_name_servers
+ * old_ip_address - for release,expire and stop
+ */
+static void process_dhclient_script_callback(const int argc, char *argv[])
+{
+    char new_ip_addr[INET_ADDRSTRLEN] = {'\0', };
+    char old_ip_addr[INET_ADDRSTRLEN] = {'\0', };
+
+    char * new_subnet_mask_len    = getenv("new_prefix"); /* IP address prefix as an integer */
+    char * new_ip_address   = getenv("new_ip_address");
+    char * new_subnet_mask = getenv("new_subnet_mask"); /* This is the address in A.B.C.D format */
+
+    char * old_subnet_mask_len    = getenv("old_prefix"); /* IP address prefix as an integer */
+    char * old_ip_address   = getenv("old_ip_address");
+    char * old_subnet_mask = getenv("old_subnet_mask"); /* This is the address in A.B.C.D format */
```

Comment:
We miss gateway and few other things being sent over DHCP.. We plan to have those added right?